### PR TITLE
[Core] add thread name to help performance profiling

### PR DESCRIPTION
**Pull request by wumuzi520**
## Why are these changes needed?

- Problem
  - When using the top - H command to observe the CPU occupation of a process, it is difficult to distinguish which thread has the highest CPU occupation
- Solution
  - Give a name to each thread.
 
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/core_worker.cc [hunk 1]
```diff
@@ -270,7 +270,8 @@ void CoreWorkerProcess::RunTaskExecutionLoop() {
   } else {
     std::vector<std::thread> worker_threads;
     for (int i = 0; i < instance_->options_.num_workers; i++) {
-      worker_threads.emplace_back([]() {
+      worker_threads.emplace_back([i] {
+        SetThreadName("worker.task" + std::to_string(i));
```
Comment (rkooo567) [inline src/ray/core_worker/core_worker.cc:274]: Just a question; Threads names don't need to globally unique right? Is it identified as (pid, thread_name)?

**Review by rkooo567 — APPROVED** [review]
LGTM Can you also show me the output of htop running with this PR?

File: src/ray/core_worker/core_worker.cc (hunk 1 above)
Comment (wumuzi520) [reply src/ray/core_worker/core_worker.cc:274]: - Yes, thread name don't need to globally unique, but sometimes I just want to see if these `worker.task` are load-balance.

**Comment by wumuzi520** [comment]
> LGTM Can you also show me the output of htop running with this PR?

The pid of the raylet is 3463.

run command `top -p 3463 -H`

image

**Review by edoakes — APPROVED** [review]
LGTM, but maybe we should prefix the names with `ray::` so it's clear they're from ray at a glance in top?

**Comment by wumuzi520** [comment]
> LGTM, but maybe we should prefix the names with `ray::` so it's clear they're from ray at a glance in top?

Because thread name has a limit of 15 characters, if `ray::` is added, the number of suffix characters will be too short to understand.

**Pull request merged**